Mathlib/RingTheory/Spectrum/Prime/Topology.lean

```diff
@@ -367,6 +367,9 @@ theorem comap_isInducing_of_surjective (hf : Surjective f) : IsInducing (comap f
 @[deprecated (since := "2024-10-28")]
 alias comap_inducing_of_surjective := comap_isInducing_of_surjective
 
+theorem comap_isEmbedding_of_surjective (hf : Surjective f) : IsEmbedding (comap f) :=
+  (isEmbedding_iff _).2 ⟨comap_isInducing_of_surjective _ _ hf, comap_injective_of_surjective f hf⟩
```

Comment:
We already have `isClosedEmbedding_comap_of_surjective` or does that only work for rings?
Or perhaps this is suggesting that the topology of `PrimeSpectrum` for semirings are wrong and the closed sets should be the vanishing sets of ring-cons instead.

Reply:
Yeah it only works for rings. Consider the mod 2 map from `Nat` to `ZMod 2`: {0} is the unique prime ideal of `ZMod 2` (a field), but its comap is not a closed point because multiples of 2 in Nat do not form a maximal ideal: Nat\{1} is the unique maximal ideal.

I think the current definition of topology on PrimeSpectrum of CommSemiring is standard, and structure sheaf, spectral space etc. work with this topology. People do consider other spectra associated to CommSemirings, for example the prime $k$-spectrum in the paper linked [here](https://leanprover.zulipchat.com/#narrow/channel/217875-Is-there-code-for-X.3F/topic/semiring.20from.20bounded.20distributive.20lattice/near/492588696).

Reply:
Can you add a comment pointing people to `isClosedEmbedding_comap_of_surjective`? Thanks.